src/cluster/managers/moderation/EventLogManager.ts

```diff
@@ -9,16 +9,46 @@ export class EventLogManager {
     public constructor(private readonly cluster: Cluster) {
     }
 
+    public async userTimedOut(member: Member): Promise<void> {
+        const channel = await this.getLogChannel('membertimeout', member.guild.id);
+        if (channel !== undefined && !await this.isExempt(member.guild.id, member.user.id) && member.communicationDisabledUntil !== null) {
+            await this.logEvent('membertimeout', channel, this.eventLogEmbed('User Was Timed Out', member.user, ModlogColour.TIMEOUT, {
+                fields: [
+                    { name: 'Until', value: `<t:${moment(member.communicationDisabledUntil).unix()}>`, inline: true }
+                ]
+            }));
+        }
+    }
+
+    public async userUnTimedOut(member: Member): Promise<void> {
+        const channel = await this.getLogChannel('memberuntimeout', member.guild.id);
+        if (channel === undefined || await this.isExempt(member.guild.id, member.id))
+            return;
+
+        const now = moment();
+        const auditEvents = await tryGetAuditLogs(member.guild, 50, undefined, AuditLogActionType.MEMBER_UPDATE);
+        const audit = auditEvents?.entries.find(e => e.targetID === member.id && moment(e.createdAt).isAfter(now.add(-1, 'second')));
```

Comment:
This might need to be revisited, the user might have been updated multiple times I guess. We should look specifically for one where the timeout was updated

Reply:
That means I need to add a check in the find condition to be sure the communicationDisabledUntil property is updated, right?

Reply:
Might be? Im not sure, we havent drilled any deeper into an audit log before besides its type. Im not sure how the details of the change are represented in there